packages/@react-spectrum/table/src/TableView.tsx

```diff
@@ -387,10 +393,12 @@ function TableVirtualizer({layout, collection, focusedKey, renderView, renderWra
   let headerHeight = layout.getLayoutInfo('header')?.rect.height || 0;
   let visibleRect = state.virtualizer.visibleRect;
 
+  let [scrollPosition, setScrollPosition] = useState(0);
   // Sync the scroll position from the table body to the header container.
   let onScroll = useCallback(() => {
     headerRef.current.scrollLeft = bodyRef.current.scrollLeft;
-  }, [bodyRef]);
+    setScrollPosition(bodyRef.current.scrollLeft);
```

Comment:
triggering a re-render for every scroll event seems like it would be really expensive.